libraries/AP_GPS/AP_GPS_DroneCAN.cpp

```diff
@@ -225,8 +225,24 @@ AP_GPS_Backend* AP_GPS_DroneCAN::probe(AP_GPS &_gps, AP_GPS::GPS_State &_state)
     return backend;
 }
 
-bool AP_GPS_DroneCAN::backends_healthy(char failure_msg[], uint16_t failure_msg_len)
+bool AP_GPS_DroneCAN::pre_arm_checks(char failure_msg[], uint16_t failure_msg_len)
 {
+    // only do these checks if there is at least one DroneCAN GPS:
+    bool found_uavan_type = false;
+    for (uint8_t i = 0; i < GPS_MAX_RECEIVERS; i++) {
+        const auto type = AP::gps().params[i].type;
+        if (type != AP_GPS::GPS_TYPE_UAVCAN &&
+            type != AP_GPS::GPS_TYPE_UAVCAN_RTK_BASE &&
+            type != AP_GPS::GPS_TYPE_UAVCAN_RTK_ROVER) {
+            continue;
+        }
+        found_uavan_type = true;
+   }
+    if (!found_uavan_type) {
+        return true;
+    }
+
+    // lint parameters and detected node IDs:
     for (uint8_t i = 0; i < GPS_MAX_RECEIVERS; i++) {
         const auto &params_i = AP::gps().params[i];
```

Comment:
maybe check for DroneCAN inside loop, and remove above?

Reply:
Done